exchange/customcachekeytest/customcachekey.json

```diff
@@ -28,16 +28,22 @@
         {
             "Type": "json",
             "TTLSeconds": 660,
-            "Key": "11_sports_22_"
+            "Key": "11_sports_22_",
+            "Data":"{ \"id\": \"appbid001\", \"impid\": \"oneImp\", \"price\": 7.64,\"cat\": [\"11_sports_22\"] }"
         }, {
             "Type": "json",
-            "TTLSeconds": 660
+            "TTLSeconds": 660,
+            "Data": "{ \"id\": \"pubbid001\", \"impid\": \"oneImp\", \"price\": 5.64,\"cat\": [\"33_news_44\"] }"
         }
     ],
     "defaultTTLs": {
         "banner": 300,
         "video": 3600,
         "audio": 1800,
         "native": 300
-    }
+    },
+    "targetDataIncludeWinners":true,
+    "targetDataIncludeBidderKeys":true,
+    "targetDataIncludeCacheBids":true,
+    "targetDataIncludeCacheVast":false
```

Comment:
The code that checks keys? I don't understand. What program exactly? Inside what function?

Reply:
lines 227+ in auction_test.go.

Reply:
Joe, these are the lines 227 and beyond of auction_test.go. What part is wrong exactly?
```224             ht[compareString].freq -= 1
225             if targData.includeCacheVast {
226                 ht[compareString].actualKeys = append(ht[compareString].actualKeys, cFound.Key)
227             }
228         }
229         for k, cachedElements := range ht {
230             if cachedElements.freq > 0 {
231                 t.Errorf("%s:  [CACHE_ERROR] Cache inconsistensy. Element %s was not expected to get cached\n", fileDisplayName, k)
232             } else if cachedElements.freq < 0 {
233                 t.Errorf("%s:  [CACHE_ERROR] Cache inconsistensy. We cached some more elements (%s) than expected.\n", fileDisplayName, k)
234             } else if targData.includeCacheVast {
235                 for i := 0; i < len(cachedElements.expectedKeys); i++ {
236                     found := false
237                     for j := 0; j < len(cachedElements.actualKeys); j++ {
238                         if strings.HasPrefix(cachedElements.expectedKeys[i], cachedElements.actualKeys[j]) {
239                             found = true
240                         }
241                     }
242                     if !found {
243                         t.Errorf("[CACHE_ERROR] Key \"%s\" was expected to be cached but was not.\n", cachedElements.expectedKeys[i])
244                     }
245                 }
246             }
247         }
248     }
249 }
```

Reply:
The old `runCacheSpec(t, fileDisplayName, specData, true, false)` function is no longer executed in `TestCustomCacheKeyJSON()` because it was completely reingeneered and its signature ended up changing to `runCacheSpec(t *testing.T, fileDisplayName string, specData *cacheSpec)` . In this new version the last two values found in the json filescarry the last two boolean values hardcoded in the old `runCacheSpec(t, fileDisplayName, specData, true, false)` . Therefore, those boolean values are now coded in the json files themselves rather than the runCacheSpec function.

Reply:
corrected boolean values on the  JSON files